landlab/bmi/bmi_bridge.py

```diff
@@ -288,46 +308,59 @@ def finalize(self):
 
         def get_var_grid(self, name):
             """Get the grid id for a variable."""
-            return 0
+            at = self._base._var_mapping[name]
+            return BMI_GRID[at]
 
         def get_var_itemsize(self, name):
             """Get the size of elements of a variable."""
-            return np.dtype("float").itemsize
+            at = self._base._var_mapping[name]
+            return self._base.grid[at][name].itemsize
 
         def get_var_nbytes(self, name):
             """Get the total number of bytes used by a variable."""
-            return self.get_itemsize(name) * self._base.grid.number_of_nodes
+            at = self._base._var_mapping[name]
+            return self._base.grid[at][name].nbytes
+            # return self.get_var_itemsize(name) * self._base.grid.number_of_nodes
 
         def get_var_type(self, name):
             """Get the data type for a variable."""
-            return str(np.dtype("float"))
+            at = self._base._var_mapping[name]
+            return str(self._base.grid[at][name].dtype)
+            # return str(np.dtype("float"))
 
         def get_var_units(self, name):
             """Get the unit used by a variable."""
             return self._cls.var_units(name)
 
         def get_value_ref(self, name):
             """Get a reference to a variable's data."""
-            return self._base.grid.at_node[name]
+            at = self._base._var_mapping[name]
+            return self._base.grid[at][name]
 
         def get_value(self, name, dest):
             """Get a copy of a variable's data."""
-            dest[:] = self._base.grid.at_node[name]
+            at = self._base._var_mapping[name]
+            dest[:] = self._base.grid[at][name]
             return dest
 
         def set_value(self, name, values):
             """Set the values of a variable."""
             if name in self.get_input_var_names():
-                if name in self._base.grid.at_node:
-                    self._base.grid.at_node[name][:] = values.flat
-                else:
-                    self._base.grid.at_node[name] = values
+                at = self._base._var_mapping[name]
+                self._base.grid[at][name][:] = values.flat
+                # if name in self._base.grid.at_node:
+                #     self._base.grid.at_node[name][:] = values.flat
+                # else:
+                #     self._base.grid.at_node[name] = values
             else:
                 raise KeyError("{name} is not an input item".format(name=name))
 
         def get_grid_origin(self, grid, origin):
             """Get the origin for a structured grid."""
-            origin[:] = (self._base.grid.node_y[0], self._base.grid.node_x[0])
+            if grid == 0:
+                origin[:] = (self._base.grid.node_y[0], self._base.grid.node_x[0])
```

Comment:
I think I understand what you are doing... basically you have set this up so that both parts of the Landlab dual graph are accessible, with the node-link-patch as the grid with id 0 and the corner-face-cell having an id of 1... 

I think this **may** make sense... but it needs documentation. 

Should one of them be default?

Reply:
Correct. BMI doesn't have a concept of a dual graph so, in BMI speak, landlab components have 2 grids: one with *nodes*, *links*, and *patches* and the other with *corners*, *faces*, and *cells*.

I don't think there can be a default. BMI requires these methods to accept a grid id as input.

Reply:
Sounds good... However, as most landlab grids are node-focused, it is almost certainly the case that users will want to use grid=0...

As long as there is documentation (I think in the class docstring is sufficient, rather than in every function), it is OK.

Reply:
@kbarnhart I've added documentation to *wrap_as_bmi* that outlines some of the differences between landlab components and BMI components.